src/lucky_record/criteria.cr

```diff
@@ -15,32 +17,50 @@ class LuckyRecord::Criteria(T, V)
   end
 
   def is(value)
-    rows.query.where(LuckyRecord::Where::Equal.new(column, V::Lucky.to_db!(value)))
+    add_where_sql_clause(LuckyRecord::Where::Equal.new(column, V::Lucky.to_db!(value)))
     rows
   end
 
-  def is_not(value)
-    rows.query.where(LuckyRecord::Where::NotEqual.new(column, V::Lucky.to_db!(value)))
-    rows
+  def not(value) : T
+    is_not(value)
   end
 
-  def gt(value)
-    rows.query.where(LuckyRecord::Where::GreaterThan.new(column, V::Lucky.to_db!(value)))
-    rows
+  def not : LuckyRecord::Criteria
+    @negate_next_criteria = true
+    self
   end
 
-  def gte(value)
-    rows.query.where(LuckyRecord::Where::GreaterThanOrEqualTo.new(column, V::Lucky.to_db!(value)))
-    rows
+  def is_not(value) : T
+    add_where_sql_clause(LuckyRecord::Where::NotEqual.new(column, V::Lucky.to_db!(value)))
   end
 
-  def lt(value)
-    rows.query.where(LuckyRecord::Where::LessThan.new(column, V::Lucky.to_db!(value)))
-    rows
+  def gt(value) : T
+    add_where_sql_clause(LuckyRecord::Where::GreaterThan.new(column, V::Lucky.to_db!(value)))
+  end
+
+  def gte(value) : T
+    add_where_sql_clause(LuckyRecord::Where::GreaterThanOrEqualTo.new(column, V::Lucky.to_db!(value)))
   end
 
-  def lte(value)
-    rows.query.where(LuckyRecord::Where::LessThanOrEqualTo.new(column, V::Lucky.to_db!(value)))
+  def lt(value) : T
+    add_where_sql_clause(LuckyRecord::Where::LessThan.new(column, V::Lucky.to_db!(value)))
+  end
+
+  def lte(value) : T
+    add_where_sql_clause(LuckyRecord::Where::LessThanOrEqualTo.new(column, V::Lucky.to_db!(value)))
+  end
+
+  private def add_where_sql_clause(sql_clause)
```

Comment:
Tiny nitpick but maybe this could be `add_clause`? I think that's still clear and a bit more concise

Reply:
I can live with that, done ✅